internal/mode/static/nginx/runtime/manager.go

```diff
@@ -94,6 +117,36 @@ func (m *ManagerImpl) Reload(ctx context.Context, configVersion int) error {
 	return nil
 }
 
+// UpdateHTTPServers uses the NGINX Plus API to update HTTP upstream servers.
+// Only usable if running NGINX Plus.
+func (m *ManagerImpl) UpdateHTTPServers(upstream string, servers []ngxclient.UpstreamServer) error {
+	if !m.IsPlus() {
+		return errors.New("cannot update HTTP upstream servers: NGINX Plus not enabled")
+	}
+
+	added, deleted, updated, err := m.ngxPlusClient.UpdateHTTPServers(upstream, servers)
+	m.logger.V(1).Info(fmt.Sprintf("Added %d upstream servers", len(added)))
+	m.logger.V(1).Info(fmt.Sprintf("Deleted %d upstream servers", len(deleted)))
+	m.logger.V(1).Info(fmt.Sprintf("Updated %d upstream servers", len(updated)))
+
+	return err
+}
+
+// GetUpstreams uses the NGINX Plus API to get the upstreams.
+// Only usable if running NGINX Plus.
+func (m *ManagerImpl) GetUpstreams() (ngxclient.Upstreams, error) {
+	upstreams, err := m.ngxPlusClient.GetUpstreams()
+	if err != nil {
+		return nil, err
+	}
+
+	if upstreams == nil {
+		return nil, errors.New("GET upstreams returned nil value")
```

Comment:
Is this an error, or does it just mean there are no upstreams?

Reply:
Looking at the function again, I don't know if it technically can be nil, since the value is initialized as empty first. I can remove this.

Reply:
Do you think we need to be defensive here and keep the nil check in case the upstream function changes?

Reply:
I think that may have been my original intention